src/nvme/ioctl.c

```diff
@@ -112,8 +112,136 @@ static void nvme_show_command(struct nvme_passthru_cmd *cmd, int err, struct tim
 	printf("timeout_ms   : %08x\n", cmd->timeout_ms);
 	printf("result       : %08x\n", cmd->result);
 	printf("err          : %d\n", err);
-	printf("latency      : %lu us\n",
-	       (end.tv_sec - start.tv_sec) * 1000000 + (end.tv_usec - start.tv_usec));
+}
+
+const char *nvme_admin_to_string(__u8 opcode)
+{
+	switch (opcode) {
+	case nvme_admin_delete_sq:
+		return "Delete I/O Submission Queue";
+	case nvme_admin_create_sq:
+		return "Create I/O Submission Queue";
+	case nvme_admin_get_log_page:
+		return "Get Log Page";
+	case nvme_admin_delete_cq:
+		return "Delete I/O Completion Queue";
+	case nvme_admin_create_cq:
+		return "Create I/O Completion Queue";
+	case nvme_admin_identify:
+		return "Identify";
+	case nvme_admin_abort_cmd:
+		return "Abort";
+	case nvme_admin_set_features:
+		return "Set Features";
+	case nvme_admin_get_features:
+		return "Get Features";
+	case nvme_admin_async_event:
+		return "Asynchronous Event Request";
+	case nvme_admin_ns_mgmt:
+		return "Namespace Management";
+	case nvme_admin_fw_commit:
+		return "Firmware Commit";
+	case nvme_admin_fw_download:
+		return "Firmware Image Download";
+	case nvme_admin_dev_self_test:
+		return "Device Self-test";
+	case nvme_admin_ns_attach:
+		return "Namespace Attachment";
+	case nvme_admin_keep_alive:
+		return "Keep Alive";
+	case nvme_admin_directive_send:
+		return "Directive Send";
+	case nvme_admin_directive_recv:
+		return "Directive Receive";
+	case nvme_admin_virtual_mgmt:
+		return "Virtualization Management";
+	case nvme_admin_nvme_mi_send:
+		return "NVMe-MI Send";
+	case nvme_admin_nvme_mi_recv:
+		return "NVMe-MI Receive";
+	case nvme_admin_dbbuf:
+		return "Doorbell Buffer Config";
+	case nvme_admin_format_nvm:
+		return "Format NVM";
+	case nvme_admin_security_send:
+		return "Security Send";
+	case nvme_admin_security_recv:
+		return "Security Receive";
+	case nvme_admin_sanitize_nvm:
+		return "Sanitize";
+	case nvme_admin_get_lba_status:
+		return "Get LBA Status";
+	default:
+		break;
+	}
+
+	return NULL;
```

Comment:
Just `return NULL` instead of `break`?

Reply:
In my understandig for the coding manner better to use the default case. By the way how about the following implementation instead? Or just I thought also to define the opcode strings as a table to use instead of the function using the switch case statements.
```c
	char *ret = NULL;

	switch (opcode) {
	case nvme_admin_delete_sq:
		ret = "Delete I/O Submission Queue";
		break;
...
	default:
		break;
	}

	return ret;
```

Reply:
I think it is terser and easier to follow to just return from the switch case, but doesn't really matter. A simple lookup table could definitely be faster. It might take up more memory unless you only stored entries up to the largest defined opcode.

Reply:
I see so will do try to use the table later.